### PR TITLE
[CLEANUP] Remove deprecated mouse events

### PR DESCRIPTION
**Pull request by btecu**

### CODE REVIEW

File: packages/@ember/-internals/views/lib/system/event_dispatcher.js [hunk 1]
```diff
@@ -311,7 +302,7 @@ export default EmberObject.extend({
       // Special handling of events that don't bubble (event delegation does not work).
       // Mimics the way this is handled in jQuery,
       // see https://github.com/jquery/jquery/blob/899c56f6ada26821e8af12d9f35fa039100e838e/src/event.js#L666-L700
-      if (MOUSE_ENTER_LEAVE_MOVE_EVENTS && EVENT_MAP[event] !== undefined) {
+      if (EVENT_MAP[event] !== undefined) {
```
Comment (simonihmig) [inline packages/@ember/-internals/views/lib/system/event_dispatcher.js:305]: You can delete the whole if-block here, as that does the special handling of these non-bubbling events, and only leave the else-block instead. The (now removed) `MOUSE_ENTER_LEAVE_MOVE_EVENTS` feature flag can effectively be seen as being always `false`, so in that way this if-block is basically dead code. Also the `EVENT_MAP` itself can be removed.

**Review by simonihmig — COMMENTED** [review]
Merging #19227 has caused some conflicts here, sorry about that! :wink:

One test is failing in CI, I haven't really been able to identify it by looking through the logs though...

File: packages/@ember/-internals/views/lib/system/event_dispatcher.js (hunk 1 above)
Comment (btecu) [reply packages/@ember/-internals/views/lib/system/event_dispatcher.js]: Thanks @simonihmig, I've removed the entire `if` block.

**Comment by rwjblue** [comment]
Due to some complications with 3.27's migration to use real modules (and the resulting massive number of deprecations being triggered) we are not 100% certain (yet) that 4.0 will be the version just after v3.28. 

In order to move ahead with v4.0 cleanup efforts (like this one), I've pushed a new branch that can serve as the target for breaking change PR's: https://github.com/emberjs/ember.js/tree/v4-cleanup. 

I've updated this PR to target that branch, can you rebase against that branch and push an update?

<details><summary>Expand for an example of the commands needed for that rebase.</summary>

```
# starting from this PR's branch

git fetch origin
git rebase origin/v4-cleanup

# fix any conflicts

# push the rebase
git push -f
```

Those steps should be _roughly_ what you need, but might need some tweaks based on your local repository setup (e.g. if you don't use `origin` for the main emberjs/ember.js repo you might have to use another remote name there).
</details>

Thank you for helping us push things forward!

**Comment by mixonic** [comment]
@btecu this PR was auto-closed when v4-cleanup was merged to master. Can you re-open for us to review/merge against master? This is excellent work and I don't want to lose it.

**Pull request closed without merging**